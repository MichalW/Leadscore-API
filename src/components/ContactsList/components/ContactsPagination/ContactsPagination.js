import PropTypes from 'prop-types';
import React, {PureComponent} from 'react';
import {Container, Pagination} from 'semantic-ui-react';

import {DEFAULT_CONTACTS_LIMIT} from '../../../../config';
import {historyType} from '../../../../constants/PropTypes';

class ContactsPagination extends PureComponent {
  onPageChange = (event, {activePage}) => {
    const {history} = this.props;

    history.push(`/contacts/${activePage}`);
  };

  render() {
    const {contactsCount} = this.props;
    const totalPages = Math.ceil(contactsCount / DEFAULT_CONTACTS_LIMIT);

    return (
      <Container textAlign="center">
        <Pagination
          defaultActivePage={1}
          totalPages={totalPages}
          onPageChange={this.onPageChange}
        />
      </Container>
    );
  }
}

ContactsPagination.propTypes = {
  contactsCount: PropTypes.number.isRequired,
  history: historyType.isRequired,
};

export default ContactsPagination;
