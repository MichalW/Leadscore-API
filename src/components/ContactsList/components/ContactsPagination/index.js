import {connect} from 'react-redux';
import {withRouter} from 'react-router-dom';

import ContactsPagination from './ContactsPagination';

const mapStateToProps = state => ({
  contactsCount: state.contacts.count,
});

export default withRouter(connect(mapStateToProps)(ContactsPagination));
