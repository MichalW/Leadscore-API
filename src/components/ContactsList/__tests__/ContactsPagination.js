import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import ContactsPagination from '../components/ContactsPagination';

it('should render ContactsPagination', () => {
  const props = {};

  const contactsPagination = shallow(<ContactsPagination {...props} />);

  expect(toJson(contactsPagination)).toMatchSnapshot();
});
