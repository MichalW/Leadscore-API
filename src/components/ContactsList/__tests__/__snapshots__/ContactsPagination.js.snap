// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ContactsPagination 1`] = `
<Route>
  <Component />
</Route>
`;
